pkg/operator/latencyprofilecontroller/profilerejectionchecker_test.go

```diff
@@ -0,0 +1,242 @@
+		},
+		{
+			name: "KCM transition from Medium -> Defaut with some configs Low",
+			activeRevisionProfiles: map[int32]configv1.WorkerLatencyProfileType{
+				9: configv1.MediumUpdateAverageReaction,
+				8: configv1.LowUpdateSlowReaction,
+				7: configv1.LowUpdateSlowReaction,
+			},
+			clusterProfile:      configv1.DefaultUpdateDefaultReaction,
+			isRejectionExpected: false,
+		},
+		{
+			name: "KCM transition from Medium -> Empty with some configs Low",
+			activeRevisionProfiles: map[int32]configv1.WorkerLatencyProfileType{
+				9: configv1.MediumUpdateAverageReaction,
+				8: configv1.LowUpdateSlowReaction,
+				7: configv1.LowUpdateSlowReaction,
+			},
+			clusterProfile:      "",
+			isRejectionExpected: false,
+		},
+		// Default/Empty -> Low rejections
+		{
+			name: "KCM transition from Default -> Low with some configs having Medium",
+			activeRevisionProfiles: map[int32]configv1.WorkerLatencyProfileType{
+				5: configv1.DefaultUpdateDefaultReaction,
+				3: configv1.MediumUpdateAverageReaction,
+			},
+			clusterProfile:      configv1.LowUpdateSlowReaction,
+			isRejectionExpected: true,
+		},
+		{
+			name: "KCM transition from Empty -> Low",
+			activeRevisionProfiles: map[int32]configv1.WorkerLatencyProfileType{
+				2: "",
+				1: "",
+			},
+			clusterProfile:      configv1.LowUpdateSlowReaction,
+			isRejectionExpected: true,
+		},
+		{
+			name: "KCM transition from Empty -> Low with some configs having Medium",
+			activeRevisionProfiles: map[int32]configv1.WorkerLatencyProfileType{
+				4: "",
+				3: configv1.MediumUpdateAverageReaction,
+			},
+			clusterProfile:      configv1.LowUpdateSlowReaction,
+			isRejectionExpected: true,
+		},
+		// Low -> Default/Empty rejections
+		{
+			name: "KCM transition from Low -> Default with some configs having Medium",
+			activeRevisionProfiles: map[int32]configv1.WorkerLatencyProfileType{
+				5: configv1.LowUpdateSlowReaction,
+				3: configv1.MediumUpdateAverageReaction,
+			},
+			clusterProfile:      configv1.DefaultUpdateDefaultReaction,
+			isRejectionExpected: true,
+		},
+		{
+			name: "KCM transition from Low -> Default",
+			activeRevisionProfiles: map[int32]configv1.WorkerLatencyProfileType{
+				5: configv1.LowUpdateSlowReaction,
+				4: configv1.LowUpdateSlowReaction,
+			},
+			clusterProfile:      configv1.DefaultUpdateDefaultReaction,
+			isRejectionExpected: true,
+		},
+		{
+			name: "KCM transition from Low -> Empty",
+			activeRevisionProfiles: map[int32]configv1.WorkerLatencyProfileType{
+				2: configv1.LowUpdateSlowReaction,
+				1: configv1.LowUpdateSlowReaction,
+			},
+			clusterProfile:      "",
+			isRejectionExpected: true,
+		},
+		{
+			name: "KCM transition from Low -> Empty with some configs having Medium",
+			activeRevisionProfiles: map[int32]configv1.WorkerLatencyProfileType{
+				4: configv1.LowUpdateSlowReaction,
+				3: configv1.MediumUpdateAverageReaction,
+			},
+			clusterProfile:      "",
+			isRejectionExpected: true,
+		},
```

Comment:
can we try day 0 here as well?

Reply:
Updated the UTs to test day-0 rejection behavior as well.